### ISSUE TITLE
LightBdd not working in Wpf but works in console

### ISSUE THREAD

**Issue by hudecekd** [LightBDD/LightBDD#244]
Hi,

I am using xunit and Li
ghtBDD together. I am executing tests by using `AssemblyRunner` and I have found out that if I execute them from Wpf application it fails somewhere deep inside net framework but I tests are used without LightBdd it works without any problem.

I have added simple app as attachment.
- works in console, fails in Wpf
- works without LightBdd

[ConsoleApp49.zip](https://github.com/LightBDD/LightBDD/files/7278161/ConsoleApp49.zip)

**Comment by Suremaker** [comment]
Hello @hudecekd,

I see you are using non-standard mechanics to run the xunit tests. I am not familiar with the `AssemblyRunner` semantics, so cannot help you too much with it.

Having said that, I have noticed that if you use the task version, i.e. `await ExecuteAsync(useTask: true)` and check `runner.Status` after awaiting for `TaskCompletionSource tcs`, the status is still Executing, and attempt to dispose the runner fails with the exception (no matter if LightBDD or pure xunit is used).
![image](https://user-images.githubusercontent.com/1894621/135913635-00abe631-223f-440c-89b4-6ffc03143448.png)

I presume that when you try to dispose the runner it is still finalizing the execution, thus you get the exception.
This is pure speculation, but it looks like the `AssemblyRunner` triggers the `OnExecutionComplete` when all tests are done but not the whole execution is finished, causing the race condition with your code that tries to dispose the runner. 

Now, why LightBDD execution fails in both cases? Well, LightBDD performs a set of operations after all tests are done, like reports generation which has to be overlapping with an attempt to dispose the runner.
I am not sure what is the proper method to properly dispose the runner here, but you can easily find out it is the race condition by delaying the disposal by adding `await Task.Delay(1000)` or similar method, after which your Wpf code does not throw. 

I hope this helps

**Comment by hudecekd** [comment]
Hi @Suremaker ,
thanks for very quick and deep response.
Actually problem I am reporting is not about using of task by switching `useTask` because its problem I have reported to xunit github https://github.com/xunit/xunit/issues/2392.
The problem with LightBDD (and I am not sure that it is problem with LightBDD but occures if LightBDD is used) is that 
- console app IS working nad wpf IS NOT.
- if I omit LightBDD and keep only xunit then console app IS working and wpf IS working.
Maybe that problem is deep in xunit but since it occurs when I use LightBDD I have reported it here first.
If you thinkg it is problem with xunit I can report it to them but I am not sure whether they will investigate it since it happens only if "another test framework such as LightBDD" is used.

Thanks agin
Dusan Hudecek
